OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs

```diff
@@ -93,7 +93,8 @@ public override Activity Tick(Actor self)
 				if (self.IsDisabled())
 					return this;
 
-				var weapon = attack.ChooseArmamentsForTarget(target, forceAttack).FirstOrDefault();
+				Armament weapon = null;
+				var attack = attackArray.FirstOrDefault(af => !af.IsTraitDisabled && (weapon = af.ChooseArmamentsForTarget(target, forceAttack).FirstOrDefault()) != null);
```

Comment:
That was for perf. Everything else I'd do would involve looking up something twice.